### PR TITLE
Support configuring liveliness in yaml files

### PR DESCRIPTION
**Pull request by ivanpauno**

### CODE REVIEW

File: include/domain_bridge/qos_options.hpp [hunk 1]
```diff
@@ -121,20 +121,47 @@ class QosOptions
    */
   DOMAIN_BRIDGE_PUBLIC
   QosOptions &
-  lifespan(const std::int64_t & lifespan);
```
Comment (adityapande-1995) [inline include/domain_bridge/qos_options.hpp:124]: Are ``deadline`` and ``lifespan`` meant to be changed ? Maybe we can leave these as const references then.

File: include/domain_bridge/qos_options.hpp [hunk 2]
```diff
@@ -87,7 +87,7 @@ class QosOptions
    * If nullopt_t is returned, then use automatic matching.
    */
   DOMAIN_BRIDGE_PUBLIC
-  std::optional<std::int64_t>
```
Comment (adityapande-1995) [inline include/domain_bridge/qos_options.hpp:90]: Should we also clean up ``std::`` in the function declarations above ? Like in ``depth(const std::size_t & depth);``

File: src/domain_bridge/domain_bridge.cpp [hunk 3]
```diff
@@ -402,9 +402,23 @@ class DomainBridgeImpl
...
-
-        qos.liveliness(qos_match.qos.liveliness());
-        qos.liveliness_lease_duration(qos_match.qos.liveliness_lease_duration());
+        if (qos_options.liveliness()) {
+          qos.liveliness(*qos_options.liveliness());
+        } else {
+          qos.liveliness(qos_match.qos.liveliness());
+        }
+        if (qos_options.liveliness_lease_duration()) {
+          const auto liveliness_lease_duration_ns = *qos_options.liveliness_lease_duration();
```
Comment (adityapande-1995) [inline src/domain_bridge/domain_bridge.cpp:411]: nit : Alignment

File: src/domain_bridge/domain_bridge.cpp [hunk 4]
```diff
@@ -402,9 +402,23 @@ class DomainBridgeImpl
...
+          if (liveliness_lease_duration_ns < 0) {
+            qos.liveliness_lease_duration(
+              rclcpp::Duration::from_nanoseconds(std::numeric_limits<std::int64_t>::max()));
```
Comment (adityapande-1995) [inline src/domain_bridge/domain_bridge.cpp:414]: nit : Alignment

File: src/domain_bridge/parse_domain_bridge_yaml_config.cpp [hunk 5]
```diff
@@ -142,6 +142,34 @@ static QosOptions parse_qos_options(YAML::Node yaml_node, const std::string & fi
...
+      auto liveliness_str = qos_node["liveliness"].as<std::string>();
+      if (0u == std::strcmp(liveliness_str.c_str(), "manual_by_topic")) {
+        options.liveliness(rclcpp::LivelinessPolicy::ManualByTopic);
+      } else if (0u == std::strcmp(liveliness_str.c_str(), "automatic")) {
+        options.liveliness(rclcpp::LivelinessPolicy::Automatic);
+      } else if (0u == std::strcmp(liveliness_str.c_str(), "system_default")) {
+        options.liveliness(rclcpp::LivelinessPolicy::SystemDefault);
+      } else {
+        throw YamlParsingError(
+                file_path, "livelines must be manual_by_topic/automatic/system_default");
```
Comment (adityapande-1995) [inline src/domain_bridge/parse_domain_bridge_yaml_config.cpp:156]: nit : Alignment

Comment (ivanpauno) [reply src/domain_bridge/parse_domain_bridge_yaml_config.cpp:156]: this is what uncrustify wanted, if not it doesn't pass.
(see that CI is passing)

File: src/domain_bridge/domain_bridge.cpp (hunk 3 above)
Comment (ivanpauno) [reply src/domain_bridge/domain_bridge.cpp:411]: what do you mean exactly?
This one looks fine, it's one line.

Uncrustify is passing.

File: src/domain_bridge/domain_bridge.cpp (hunk 4 above)
Comment (ivanpauno) [reply src/domain_bridge/domain_bridge.cpp:414]: same, this one looks fine to me

File: include/domain_bridge/qos_options.hpp (hunk 1 above)
Comment (ivanpauno) [reply include/domain_bridge/qos_options.hpp:124]: There's no reason to pass integers as `const &`.
I think it's a good idea to change it now, it shouldn't affect domain bridge users.

File: include/domain_bridge/qos_options.hpp (hunk 2 above)
Comment (ivanpauno) [reply include/domain_bridge/qos_options.hpp:90]: yes

Comment (ivanpauno) [reply include/domain_bridge/qos_options.hpp:90]: Fixed in https://github.com/ros2/domain_bridge/pull/73/commits/4e66a4fad7e145e9beae59831c1b66c9d3869ef7

**Review by adityapande-1995 — APPROVED** [review]
Tested locally, works !

**Comment by ivanpauno** [comment]
Thanks for reviewing @adityapande-1995 !!!

**Pull request merged**